src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -119,6 +129,7 @@ private enum Flags : byte
             _refKind = refKind;
             _modifiers = modifiers;
             _explicitInterfaceType = explicitInterfaceType;
+            _diagnostics = diagnostics;
```

Comment:
>_diagnostics = diagnostics; [](http://example.com/codeflow?start=12&length=27)

This is a wrong thig to do. We cannot cache BindingDiagnosticBag that is passed to the constructor. Someone else controls the lifetime of this bag. Since we are likely to return this bag back into the pool soon after the constructor returns, later on we will "pollute" this bag when someone else is using it. Quite possibly a different compilation.